client/state/index.js

```diff
@@ -111,7 +112,9 @@ export const reducer = combineReducers( {
 	ui,
 	users,
 	wordads,
-} );
+};
+
+export const reducer = combineReducers( { ...extensions.reducers(), ...reducers } );
```

Comment:
Well, I was going off the original PR (#8103) as a baseline. However, I'd be up for something better here. We could automatically take care of this as part of the static build. I see two options:
1. Prefix all extension reducers. (e.g. `extension_{$extensionName}`)
2. Combine all extension reducers under a single one named `extensions`